webapp/app/services/roleService.js

```diff
@@ -1,17 +1,38 @@
 define('services/roleService', [], function () {
 
-    function roleReadAll() {
-    // TODO: refactor common headers and base url
-    return $.ajax({
-      type: 'POST',
-      contentType: 'application/json',
-      url:'http://localhost:10160/Service.svc/jsonp/post',
-      data: '{"requests":[{"__type":"RoleReadAllRequest:#P202.Training.WCF.RequestsAndResponses"}]}'
-    });
-  }
+    function roleReadAll() {        
+        return $.ajax({
+            type: 'POST',
+            contentType: 'application/json',
+            url: 'http://localhost:10160/Service.svc/jsonp/post',
+            data: '{"requests":[{"__type":"RoleReadAllRequest:#P202.Training.WCF.RequestsAndResponses"}]}'
+        });
+    }
 
-  return {
-      roleReadAll: roleReadAll
-  };
+    function roleAdd(roleName) {
+        var role = {"Name":roleName};              
```

Comment:
Do you need this auxiliary variable? perhaps you can solve the stringify without that

Reply:
i move the object role to stringify